media/javascript/rtd-import.js

```diff
@@ -31,6 +31,72 @@
 
       repo.val(type);
     });
+
+    $('[data-sync-repositories]').each(function () {
+      var $button = $(this);
+      var target = $(this).attr('data-target');
+
+      $button.on('click', function () {
+        var url = $button.attr('data-sync-repositories');
+        $.ajax({
+          method: 'POST',
+          url: url,
+          success: function (data) {
+            $button.attr('disabled', true);
+            watchProgress(data.url);
+          },
+          error: function () {
+            onError();
+          }
+        });
+        $('.sync-repositories').addClass('hide');
+        $('.sync-repositories-progress').removeClass('hide');
+      });
+
+      function watchProgress(url) {
+        setTimeout(function () {
+          $.ajax({
+            method: 'GET',
+            url: url,
+            success: function (data) {
+              if (data.finished) {
+                if (data.success) {
+                  onSuccess();
+                } else {
+                  onError();
+                }
+              } else {
+                watchProgress(url);
+              }
+            },
+            error: function () {
+              watchProgress(url);
+            }
+          });
+        }, 2000);
+      }
```

Comment:
This should include a timeout to display a "can't sync" error. My install executes the task fine, but doesn't seem to update the client correctly, which keeps me waiting forever for a sync to finish